### PR TITLE
Fix missing theming for login button

### LINKED ISSUE #27495: "Log in" Button is not Themed
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install NC 21.0.2.1
2. Open a login page
3. Move the cursor to the "Log in" button

### Expected behaviour
"Log in"' button is themed with colour ``#B64A44``.

### Actual behaviour
"Log in" button turns into default colour ``#0082C9``.
login

### Server configuration

**Operating system:** Alpine 3.13

**Web server:** nginx/1.18.0

**Database:** mysql  Ver 15.1 Distrib 10.5.9-MariaDB, for Linux (armv8l) using readline 5.1

**PHP version:** 7.4.19

**Nextcloud version:** (see Nextcloud admin page) 21.0.2.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 21.0.1.1

**Where did you install Nextcloud from:** Nextcloud official / updater

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.7.0
  - admin_audit: 1.11.0
  - audioplayer: 3.1.0
  - bookmarks: 4.2.2
  - bruteforcesettings: 2.1.0
  - calendar: 2.2.2
  - checksum: 1.1.2
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_external: 1.12.0
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - news: 15.4.5
  - nextcloud_announcements: 1.10.0
  - notes: 4.0.4
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - passwords: 2021.6.20
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - recommendations: 1.0.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - tasks: 0.13.6
  - text: 3.2.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - activity
  - encryption
  - federation
  - files_trashbin
  - issuetemplate
  - sharebymail
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
           "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "21.0.2.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "default_language": "ja_JP",
        "default_locale": "ja_JP",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "maintenance": false,
        "theme": ""
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 78.11.0esr

**Operating system:** Debian 10.9

#### Web server error log
<details>
<summary>Web server error log</summary>

```
(Not an error.)
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
(Not an error.)
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
(Not an error.)
```
</details>

### PR DESCRIPTION
**Pull request by CarlSchwan**
Close #27495

image

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Comment by szaimen** [comment]
Wow, thank you very much for so many PRs, @CarlSchwan ! :)

File: apps/theming/css/theming.scss [hunk 1]
```diff
@@ -168,6 +168,16 @@ input.primary,
 	color: $color-primary-text;
 }
 
+input.primary:not(:disabled):hover,
```
Comment (juliushaertl) [inline apps/theming/css/theming.scss:171]: I think we should probably wrap this with in `#body-login {}` as the gradient is only used on the login page as far as I can see.

File: apps/theming/css/theming.scss [hunk 2]
```diff
@@ -168,6 +168,18 @@ input.primary,
 	color: $color-primary-text;
 }
 
+#body-login {
+	input.primary:not(:disabled):hover,
+	input.primary:not(:disabled):focus,
+	button.primary:not(:disabled):hover,
+	button.primary:not(:disabled):focus,
+	a.primary:not(:disabled):hover,
+	a.primary:not(:disabled):focus {
```
Comment (Pytal) [inline apps/theming/css/theming.scss:177]:
```suggestion
	input.primary:enabled:hover,
	input.primary:enabled:focus,
	button.primary:enabled:hover,
	button.primary:enabled:focus,
	a.primary:enabled:hover,
	a.primary:enabled:focus {
```
More concise :)

**Review by jancborchardt — APPROVED** [review]
With the gradient, nice work! :)

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Sign-off message is missing on the second commit :wink:

**Review by Pytal — APPROVED** [review]

**Pull request merged**